qiskit/compiler/transpiler.py

```diff
@@ -372,6 +404,7 @@ def _transpile_circuit(circuit_config_tuple: Tuple[QuantumCircuit, Dict]) -> Qua
 
     # we choose an appropriate one based on desired optimization level
     level = transpile_config["optimization_level"]
+    shared_transpiler_args["optimization_level"] = optimization_level
```

Comment:
So playing with this locally I see what I was doing here. I wanted to drop it from the input shared args so I can just do `PassManagerConfig(**shared_transpiler_args)` without a copy but then needed to restore it so that it wasn't missing from subsequent uses. I've simplified the code a bit to make this clearer.

Reply:
Simplified in: https://github.com/Qiskit/qiskit-terra/pull/7789/commits/bb2f8e223139f5eea49cfa022f92518a76ca56de